src/pip/_internal/utils/glibc.py

```diff
@@ -10,9 +10,37 @@
     from typing import Optional, Tuple
 
 
-def glibc_version_string():
+def glibc_version_string_os():
     # type: () -> Optional[str]
-    "Returns glibc version string, or None if not using glibc."
+    """Returns glibc version string, or None if not using glibc.
+
+    This should be paired with glibc_version_string_ctypes as a fallback:
+    glibc_ver = glibc_version_string_os() or glibc_version_string_ctypes()
+    """
+
+    # os.confstr is quite a bit faster than ctypes.DLL.
+    # It's also less likely to be broken or missing.
+
+    # This strategy is used in the standard library platform module:
+    # https://github.com/python/cpython/blob/fcf1d003bf4f0100c9d0921ff3d70e1127ca1b71/Lib/platform.py#L175-L183
+
```

Comment:
Just a quick note: you don't need any of these three empty lines. Also, the three sentences can follow each other like a paragraph, with the link on its own line at the end.